### PR TITLE
add nfts selection in LLD CLS

### PR DESCRIPTION
**Pull request by pierrelouis-c**
### 📝 Description

This PR adds NFTs selection in LLD Custom lock screen.

### ❓ Context

- **Impacted projects**: `ledger-live-desktop` 
- **Linked resource(s)**: [Figma](https://www.figma.com/file/Le3nxsc4L8ApwHXFbMbPsA/LLD-%2F-Onboarding-%2F-Stax?node-id=90%3A26895&t=m3U9S96wVvCyvfEg-0), [FAT-803] 

### ✅ Checklist

- [x] **Test coverage** (includes partial e2e coverage with screenshot testing of the NFT gallery view in 2 states: populated and empty state. We cannot go further in the flow as we don't have an easy way to mock NFT metadata retrieval)
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://user-images.githubusercontent.com/91890529/217511764-dffd90b1-2294-4b2e-a47d-00d704e3a1e3.mp4

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[FAT-803]: https://ledgerhq.atlassian.net/browse/FAT-803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/nft/Gallery/TokensList/Item.jsx [hunk 1]
```diff
@@ -60,9 +60,17 @@ type Props = {
   mode: "grid" | "list",
   withContextMenu?: boolean,
   onHideCollection?: () => void,
+  overloadOnItemClick?: (nftMetadata: NFTMetadata) => void,
```
Comment (thomasrogerlux) [inline apps/ledger-live-desktop/src/renderer/screens/nft/Gallery/TokensList/Item.jsx:63]: It's override not overload eheh

Also I would just have a `onItemClick?: (nftMetadata: NFTMetadata)` and rename the default one to `defaultOnItemClick`

File: apps/ledger-live-desktop/src/renderer/screens/nft/Gallery/TokensList/Item.jsx [hunk 2]
```diff
@@ -94,14 +102,14 @@ const NftCard = ({ id, mode, account, withContextMenu = false, onHideCollection
...
-        onClick={onItemClick}
+        onClick={overloadOnItemClick ? () => overloadOnItemClick(metadata) : onItemClick}
       >
         <Skeleton width={40} minHeight={40} full={isGrid} show={show}>
           <Media
             metadata={metadata}
             tokenId={nft.tokenId}
-            size={40}
-            full={isGrid}
+            size={overloadOnItemClick ? 180 : 40}
```
Comment (thomasrogerlux) [inline apps/ledger-live-desktop/src/renderer/screens/nft/Gallery/TokensList/Item.jsx:111]: It seems weird to link the onClick overloading to style properties, separate it instead

File: apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx [hunk 3]
```diff
@@ -1,19 +1,100 @@
...
+  }, [currentBase64, onError, onResult]);
+
+  const handleClickPrevious = useCallback(() => {
+    setCurrentBase64(null);
+    setIsShowingNftGallery(false);
+  }, []);
+
+  const handlePickNft = useCallback(
+    (nftMetadata: NFTMetadata) => {
+      const uri = extractNftBase64(nftMetadata);
```
Comment (thomasrogerlux) [inline apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx:53]: You can use the function `urlContentToDataUri()` from Olivier in `src/renderer/components/CustomImage/shared.ts` to save some logic here

File: apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx [hunk 4]
```diff
@@ -1,19 +1,100 @@
...
+    <StepContainer
+      footer={
+        isShowingNftGallery ? (
+          <StepFooter
+            nextDisabled={!currentBase64}
+            nextLabel={t("customImage.steps.choose.selectNft")}
+            previousLabel={t("customImage.steps.choose.previousSelectNft")}
+            onClickNext={handleClickNext}
+            onClickPrevious={handleClickPrevious}
+            nextTestId="custom-image-select-nft-button"
```
Comment (thomasrogerlux) [inline apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx:83]: Isn't it `dataTestId` ?

File: apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx [hunk 5]
```diff
@@ -1,19 +1,100 @@
...
 
 type Props = StepProps & {
   onResult: (res: ImageBase64Data) => void;
   setLoading: (_: boolean) => void;
 };
 
+const extractNftBase64 = (metadata: NFTMetadata) => {
+  const mediaTypes = metadata ? getMetadataMediaTypes(metadata) : null;
+  const mediaSizeForCustomImage = mediaTypes
+    ? ["original", "big", "preview"].find(size => mediaTypes[size] === "image")
```
Comment (thomasrogerlux) [inline apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx:26]: Can you move the `original`, `big` etc to a variable outside the component or get those from somewhere if it's generic values?

**Review by thomasrogerlux — COMMENTED** [review]
Sorry eheh

File: apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx (hunk 4 above)
Comment (pierrelouis-c) [reply apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx:83]: Yes it is data-test-id for playwright but nextTestId will set the id in the Footer next button. Not directly in this component.

**Review by thomasrogerlux — APPROVED** [review]

**Pull request merged**